### PR TITLE
Meta: add infrastructure to generate summary.json

### PR DESCRIPTION
**Pull request by annevk**
The setup is as follows:

1. We collect all the issue data from the repository through the GitHub API and store it in a JSON file. Storing the input allows us to monitor changes in it for bugs. We'll likely want to change some issues so they are more suitable for consumption by the script.
2. The script takes this data and produces a JSON file with a summarized view of each issue. The bulk of the logic is in processing the body of the issue (the top comment) which is split between before and after the YAML change.

The idea is that the JSON file with the summarized view can be consumed and eventually displayed on webkit.org once we're happy with it.

It currently has some flaws due to the raw data, but it seems worthwhile to check this in and then improve the raw data to validate the entire process around updating the issue data.

### CODE REVIEW

File: summary.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    parser.add_argument("-p", "--process", action="store_true", help="process the data")
+    args = parser.parse_args()
+
+    if args.update:
+        # GitHub allows us to read issues in increments of 100, called pages. As we don't have more
+        # than 3 pages we're not optimizing this for now.
+        data = []
+        page = 1
+        while True:
+            temp_data = json.loads(requests.get(f"https://api.github.com/repos/WebKit/standards-positions/issues?direction=asc&state=all&per_page=100&page={page}").text)
```
Comment (hober) [inline summary.py:121]: should probably pass in a timeout duration & handle it

**Review by hober — APPROVED** [review]
looks good to me; one nit

File: summary.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    write_json("summary.json", summary)
+
+def is_ignorable_issue(issue):
+    if "pull_request" in issue:
+        return True
+    for label in issue["labels"]:
+        if label["name"] == "meta":
+            return True
+        elif label["name"] == "proposal withdrawn":
+            return True
```
[the remark this answers is not in the document]
Comment (annevk) [reply summary.py]: Heh, thanks! I had made a similar change locally (though using a tuple) when I needed to add "`invalid`".

File: summary.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        write_json("summary-data.json", data)
+        print("Done, thanks for updating!")
+        exit(0)
+    
+    if args.process:
+        if not os.path.exists("summary-data.json"):
+            print("Sorry, you have to update first.")
+            exit(1)
+
+        data = json.load(open("summary-data.json", "r"))
```
Comment (gsnedders) [inline summary.py:141]:
```suggestion
        with open("summary-data.json", "rb") as f:
            data = json.load(f)
```

Similarly, close the file. Also avoid decoding the file using whatever encoding Python is configured to use, and let the JSON decoder just figure it out.

**Review by gsnedders — COMMENTED** [review]
Here's a lot of comments.

You might want to run `black` on this to reformat this to meet expected code style, and you may also want to run `flake8` on this and see what errors it finds?

**Review by karlcow — CHANGES_REQUESTED** [review]
Currently the code resides in the root directory. 
It would probably be better in a separate directory. 
Screenshot 2023-06-23 at 14 49 39


Specifically when adding a couple of things for managing the tests, etc. 

Could we move all of that in a tooling or something.

Also should  the edits on `summary.py` be separated from `summary.json` and `summary-data.json`

I would propose we do. 

```
.github/
data/
   summary-data.json
   summary.json
tooling/
    summary.py
    tests/
```

I will be working on testing later. Once the first PR is merged.

**Comment by karlcow** [comment]
> Also should the edits on `summary.py` be separated from `summary.json` and `summary-data.json`

It could even be part of a GitHub actions. So that we have a clear separation in between editing the python factory and the data generations.

**Comment by gsnedders** [comment]
> It could even be part of a GitHub actions.

One could reasonably imagine the `summary.json` getting pushed to `gh-pages`, so it could be used by others.

File: summary.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                response = requests.get(f"https://api.github.com/repos/WebKit/standards-positions/issues?direction=asc&state=all&per_page=100&page={page}", timeout=5)
+            except:
+                print("Updated failed, network failure or request timed out.")
+                exit(1)       
+            temp_data = json.loads(response.text)
+            if len(temp_data) > 0:
+                data.extend(temp_data)
+                page += 1
+                continue
+            break
```
[the remark this answers is not in the document]
Comment (annevk) [reply summary.py:131]: I don't think it's that simple. I could replace the continue/break with an else/break branch (and then perhaps flip the branches), but I kinda liked this personally.

File: summary.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (annevk) [reply summary.py:1]: Is it really best practice to put the version number there? It works fine either way.

[the remark this answers is not in the document]
Comment (karlcow) [reply summary.py:1]: Let's assume this is python 3, and we can set a requirements after.

**Comment by karlcow** [comment]
> I like the idea of more automation, but changing `summary.json` through PRs is important for review purposes in my opinion. We want to be somewhat careful about what ends up on the website. I could see running the update process in a more automated fashion, but for now that seems more wasteful than useful to me.

At least would it be possible to do different PRs for python code and data file generation. 🙏

**Comment by annevk** [comment]
I doubt we're going to change the code a whole lot after this lands and a reviewer probably wants to see the changes doing the right thing?

**Pull request merged**